From 7c310c9b0439a020add4c23d84ab33eeb486bd75 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 30 Oct 2024 20:32:25 +0200
Subject: [PATCH 1/2] libtool.m4: Update hardcode_shlibpath_var for OpenBSD

When generating an executable in OpenBSD, the executable fails to
generate when shlibpath_var, LD_LIBRARY_PATH, is used solely to
locate dependent libraries.

For hardcoding library paths in OpenBSD, see documentation for
hardcode_direct and hardcode_libdir_flag_spec.

* m4/libtool.m4: Update hardcode_shlibpath_var from 'no' to 'unsupported'
  by removing 'with_gnu_ld=no' for OpenBSD.
---
 m4/libtool.m4 | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 5092b0e3..857e9c11 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -5091,9 +5091,6 @@ dnl Note also adjust exclude_expsyms for C++ above.
     # we just hope/assume this is gcc and not c89 (= MSVC++ or ICC)
     with_gnu_ld=yes
     ;;
-  openbsd*)
-    with_gnu_ld=no
-    ;;
   esac
 
   _LT_TAGVAR(ld_shlibs, $1)=yes
-- 
2.45.2

